### DISCUSSION TITLE
Implementating a batch text replacements - extension necessary?

### DISCUSSION THREAD

**Discussion by jgradatwork** [microsoft/vscode-discussions#2446] [Extension Development QnA] [answered]
Greetings friends,

I'm pretty new to VSCode and I've been away from coding for almost 2 decades now, but I'm trying to get back into the swing of things. I'd like to pick your brain on my current problem:

I have a very limited need for a batch text replacement solution that would like to run a (longish, current ~5000) series of possible text replacements on a given text in VSC, all currently defined in a text file with "=" as delimiter between text_to_be_replace and text_to_replace_with. Since I have to do it at work and can't install the proper VSC, I can't use any of the published text manipulation extensions that would allow me to do this, so I'm trying to figure out what my options are.

Does anyone have advice on how to implement this in VSC directly? Is there a basic function I'm not aware of that would make this less painful? Is writing an extension my best bet here?

Thanks a lot for chiming in!

**Comment by alefragnani** [comment, marked as answer]
Hi @jgradatwork ,

I'm not sure what you mean about "proper VSC", but since VS Code does not provide a built in _macro recording_ feature, there is no way to achieve this using _just_ VS Code. Even if you can't install published extensions, you will only achieve this using extensions. Either published or made "at work", and installed manually, using its `.vsix` file.

But, if (even at work) extensions are out of questions, I see no other way than you building a customized version of VS Code, and _embed_ a built in extension that does this. Doing so, your VS Code version will have the feature "out of the box".

On the other hand, I'm not sure you noticed the recent VS Code release. In 1.96, the VS Code team provided the concept of [Allowed List of Extensoins](https://code.visualstudio.com/updates/v1_96#_configure-allowed-extensions), with Enterprise users in mind. Maybe you could show this to the IT guys at work, and you could be allowed to install some extensions that already does what you need.

Hope this helps
